code/game/turfs/simulated/floor/plating.dm

```diff
@@ -193,30 +193,32 @@
 		SSobj.processing |= src
 
 /turf/simulated/floor/plating/lava/process()
-	if(!contents)
+	if(!burn_stuff())
 		processing = 0
 		SSobj.processing.Remove(src)
-		return
-	burn_stuff()
+	
 
 /turf/simulated/floor/plating/lava/proc/burn_stuff()
+	. = 0
 	for(var/atom/movable/AM in contents)
```

Comment:
Yes MSO, which is why you do ```/atom```, the AM part is unused, since all code is under ```istype(obj)``` or ```istype(mob/living)```

And this won't increase the number of objects looped through either, since byond already loops through them all, it just has to ```istype()``` them which is unneeded here